[BUG] NetBSD 11.0 RC4 and generally NetBSD 11.0 support

**Describe the bug**
Unable to complile on NetBSD 11.0 RC4

**To Reproduce**
Download
configure fails
```
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for x86_64--netbsd-gcc... (cached) gcc
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to enable C11 features... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
configure: error: Unknown NetBSD release: 11.0_RC4
*** Error code 1

Stop.
make[1]: stopped making "all" in /srv/NetBSD/pkgsrc/wip/lsof
*** Error code 1
```

**Expected behavior**
Add compatibility to NetBSD 11 and sucessfully compile

**Program output**
N/A

**Environment (please complete the following information):**

 - Kernel: 11.0RC4
 - OS: NetBSD
 - lsof Version: lsof-4.99.6
 - Origin: tarball

**Additional context**
Please fix :-)

--- a/Configure
+++ b/Configure
@@ -2662,10 +2662,15 @@ return(0); }
 	echo "!!!WARNING!!!  Unsupported NetBSD version: $LSOF_VSTR"
 	echo "!!!WARNING!!!  Configuring for NetBSD 10.0"
 	;;
+      11.*)
+	LSOF_VERS="11000000"
+	echo "!!!WARNING!!!  Unsupported NetBSD version: $LSOF_VSTR"
+	echo "!!!WARNING!!!  Configuring for NetBSD 11.0"
+	;;
       *)
 	echo "Unknown NetBSD release: $LSOF_VSTR"
-	echo Assuming NetBSD 10.0
-	LSOF_VERS="10000000"
+	echo Assuming NetBSD 11.0
+	LSOF_VERS="11000000"
 	;;
       esac	# }
     fi	# }

--- a/configure.ac
+++ b/configure.ac
@@ -466,7 +466,7 @@ AC_CHECK_DECLS([kvm_getproc2], [], [], [#include <kvm.h>])
 # Detect fs/ptyfs/ptyfs.h header for HASPTYFS
 AC_CHECK_HEADERS([fs/ptyfs/ptyfs.h])
 
-# Detect fs/ptyfs/ptyfs.h header for HASPROCFS
+# Detect miscfs/procfs/procfs.h header for HASPROCFS
 AC_CHECK_HEADERS([miscfs/procfs/procfs.h])
 
 # Detect PFSroot enum for HASPROCFS_PFSROOT
@@ -487,6 +487,8 @@ AS_IF([test x$LSOF_DIALECT = xnetbsd], [
 
 	AC_MSG_NOTICE([lockf.h creation succeeded])
 	CFLAGS="$CFLAGS -DHAS_LOCKF_H"
+	# Hardcode some flags for latest NetBSD versions
+	CFLAGS="$CFLAGS -DHASPROCFS_PFSROOT"
 ])
 
 # Detect fdescfs version for HASFDESCFS

--- a/configure.ac
+++ b/configure.ac
@@ -134,6 +134,11 @@ AS_CASE([${host_os}],
 			AC_MSG_WARN([Unsupported NetBSD release: $(uname -r)])
 			AC_MSG_WARN([Configuring for NetBSD 10.0])
 			;;
+		11.*)
+			LSOF_VERS="11000000"
+			AC_MSG_WARN([Unsupported NetBSD release: $(uname -r)])
+			AC_MSG_WARN([Configuring for NetBSD 11.0])
+			;;
 		*)
 			AC_MSG_ERROR([Unknown NetBSD release: $(uname -r)])
 			;;

--- a/lib/dialects/netbsd/Mksrc
+++ b/lib/dialects/netbsd/Mksrc
@@ -26,7 +26,12 @@ L="dlsof.h dmnt.c dnode.c dnode1.c dproc.c dproto.h dsock.c dstore.c machine.h"
 
 mksrc
 
+D=lib
+L="node.c"
+
+mksrc
+
 D=src
-L="arg.c main.c node.c print.c ptti.c store.c usage.c util.c"
+L="arg.c main.c print.c ptti.c store.c usage.c util.c"
 
 mksrc

--- a/lib/dialects/netbsd/dlsof.h
+++ b/lib/dialects/netbsd/dlsof.h
@@ -498,9 +498,9 @@ struct mounts {
 extern int Np; /* number of kernel processes */
 
 #    if defined(HASKVMGETPROC2)
-struct kinfo_proc2 *P; /* local process table copy */
+extern struct kinfo_proc2 *P; /* local process table copy */
 #    else              /* ! defined(HASKVMGETPROC2) */
-struct kinfo_proc *P;      /* local process table copy */
+extern struct kinfo_proc *P;      /* local process table copy */
 #    endif             /* defined(HASKVMGETPROC2) */
 
 extern int pgshift; /* kernel's page shift */

--- a/lib/dialects/netbsd/dnode.c
+++ b/lib/dialects/netbsd/dnode.c
@@ -439,7 +439,7 @@ process_overlaid_node:
             ksb.st_ino = (ino_t)2;
             ksb.st_size = DEV_BSIZE;
             ksbs = 1;
-        } else if (Namech[0] && statsafely(Namech, &ksb) == 0)
+        } else if (Namech[0] && statsafely(ctx, Namech, &ksb) == 0)
             ksbs = 1;
         nty = KERNFSNODE;
         break;

--- a/lib/rnmt.c
+++ b/lib/rnmt.c
@@ -137,13 +137,13 @@ static void ncache_walk(struct lsof_context *ctx, KA_T ncp,
     if (sanity_check_vnode_impl(&vi) == 0 && sanity_check_namecache(&nc) == 0) {
         lnc = ncache_enter_local(vp, plnc, &nc);
         if (vi.vi_vnode.v_type == VDIR && vi.vi_nc_tree.rbt_root != NULL) {
-            ncache_walk((KA_T)vi.vi_nc_tree.rbt_root, lnc);
+            ncache_walk(ctx, (KA_T)vi.vi_nc_tree.rbt_root, lnc);
         }
     }
     if (left)
-        ncache_walk(left, plnc);
+        ncache_walk(ctx, left, plnc);
     if (right)
-        ncache_walk(right, plnc);
+        ncache_walk(ctx, right, plnc);
 }
 
 void ncache_load(struct lsof_context *ctx) {
@@ -151,7 +151,7 @@ void ncache_load(struct lsof_context *ctx) {
     struct vnode_impl vi;
 
     rootvnode_addr = (KA_T)0;
-    if (get_Nl_value("rootvnode", (struct drive_Nl *)NULL, &rootvnode_addr) <
+    if (get_Nl_value(ctx, "rootvnode", (struct drive_Nl *)NULL, &rootvnode_addr) <
             0 ||
         !rootvnode_addr ||
         kread(ctx, (KA_T)rootvnode_addr, (char *)&rootvnode_addr,
@@ -161,7 +161,7 @@ void ncache_load(struct lsof_context *ctx) {
     }
 
     rb_tree_init(&lnc_rbtree, &lnc_rbtree_ops);
-    ncache_walk((KA_T)vi.vi_nc_tree.rbt_root, 0);
+    ncache_walk(ctx, (KA_T)vi.vi_nc_tree.rbt_root, 0);
 }
 
 static void build_path(char **buf, size_t *remaining, const struct lnc *lnc) {
@@ -187,7 +187,7 @@ static void build_path(char **buf, size_t *remaining, const struct lnc *lnc) {
     *buf += len;
 }
 
-char *ncache_lookup(char *buf, int blen, int *fp) {
+char *ncache_lookup(struct lsof_context *ctx, char *buf, int blen, int *fp) {
     const struct lnc *lnc;
     char *p;
     size_t remaining;

--- a/lib/rnmt.c
+++ b/lib/rnmt.c
@@ -39,6 +39,11 @@ struct lnc {
     char lnc_name[NCHNAMLEN + 1]; /* name */
 };
 
+/* For NetBSD 10 compat */
+#ifndef NC_NLEN
+#define	NC_NLEN(ncp)	((ncp)->nc_nlen)
+#endif
+
 static const rb_tree_ops_t lnc_rbtree_ops = {
     .rbto_compare_nodes = lnc_compare_nodes,
     .rbto_compare_key = lnc_compare_key,
@@ -84,7 +89,7 @@ static struct lnc *ncache_enter_local(KA_T vp, const struct lnc *plnc,
     }
     lnc->lnc_vp = vp;
     lnc->lnc_plnc = plnc;
-    lnc->lnc_nlen = nc->nc_nlen;
+    lnc->lnc_nlen = NC_NLEN(nc);
     memcpy(lnc->lnc_name, nc->nc_name, lnc->lnc_nlen);
     lnc->lnc_name[lnc->lnc_nlen] = 0;
 
@@ -104,13 +109,13 @@ static int sanity_check_namecache(const struct namecache *nc) {
     if (nc->nc_vp == NULL)
         return -1;
 
-    if (nc->nc_nlen > NCHNAMLEN)
+    if (NC_NLEN(nc) > NCHNAMLEN)
         return -1;
 
-    if (nc->nc_nlen == 1 && nc->nc_name[0] == '.')
+    if (NC_NLEN(nc) == 1 && nc->nc_name[0] == '.')
         return -1;
 
-    if (nc->nc_nlen == 2 && nc->nc_name[0] == '.' && nc->nc_name[1] == '.')
+    if (NC_NLEN(nc) == 2 && nc->nc_name[0] == '.' && nc->nc_name[1] == '.')
         return -1;
 
     return 0;

--- a/tests/LTbasic2.c
+++ b/tests/LTbasic2.c
@@ -117,9 +117,17 @@ int main(int argc, char **argv) {
                     }
                 } else if (f->fd_type == LSOF_FD_NUMERIC) {
                     /* check if fd matches */
-                    if (f->fd_num == fd && f->name &&
-                        strstr(f->name, "LTbasic2-tmp")) {
+                    if (f->fd_num == fd && f->inode == tmp_stat.st_ino &&
+                        f->dev == tmp_stat.st_dev) {
+#ifdef NETBSDV
+                        /* on NetBSD, the file path for new created file may be
+                         * unavailable, skip validation */
                         fd_found = 1;
+#else
+                        if (f->name && strstr(f->name, "LTbasic2-tmp")) {
+                            fd_found = 1;
+                        }
+#endif
                     }
                 }
             }

--- a/tests/LTsock.c
+++ b/tests/LTsock.c
@@ -631,11 +631,11 @@ static char *FindSock(int fn) /* function -- an LT_FBY* value */
      * Complete the option vector and start lsof execution.
      */
 
+    mk_end = ti;
 #if defined(USE_LSOF_C_OPT)
     opv[ti++] = "-C";
 #endif /* defined(USE_LSOF_C_OPT) */
 
-    mk_end = ti;
     opv[ti] = (char *)NULL;
     if ((cem = ExecLsof(opv))) {
         for (tj = mk_start; tj < mk_end; tj++)

